extensions/arc/src/ui/dialogs/connectControllerDialog.ts

```diff
@@ -149,3 +175,51 @@ export class ConnectToControllerDialog extends InitializingComponent {
 		return this._completionPromise.promise;
 	}
 }
+
+export class PasswordToControllerDialog extends ConnectToControllerDialog {
+
+	constructor(protected _treeDataProvider: AzureArcTreeDataProvider, title: string = loc.passwordToController) {
+		super(_treeDataProvider, title);
+	}
+
+	public async validate(): Promise<boolean> {
+		if (!this.passwordInputBox.value) {
+			return false;
+		}
+		const azdataApi = <azdataExt.IExtension>vscode.extensions.getExtension(azdataExt.extension.name)?.exports;
+		try {
+			await azdataApi.azdata.login(this.urlInputBox.value!, this.usernameInputBox.value!, this.passwordInputBox.value);
+		} catch (e) {
+			if (getErrorMessage(e).match(/Wrong username or password/i)) {
+				this.dialog.message = {
+					text: loc.invalidPassword,
```

Comment:
Why aren't you just printing the error message directly? azdata should be providing us with a useful error message I'd hope...